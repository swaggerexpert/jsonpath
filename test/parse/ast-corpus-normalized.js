import { assert } from 'chai';
import { jestExpect as expect } from 'mocha-expect-snapshot';

import { parse, ASTTranslator } from '../../src/index.js';

describe('parse', function () {
  context('ast-corpus-normalized', function () {
    const normalizedPaths = [
      // https://www.rfc-editor.org/rfc/rfc9535#section-2.7.1
      "$['a']",
      '$[1]',
      '$[2]',
      "$['a']['b'][1]",
      "$['\\u000b']",
    ];

    normalizedPaths.forEach((normalizedPath) => {
      specify(normalizedPath, function () {
        const parseResult = parse(normalizedPath, {
          normalized: true,
          translator: new ASTTranslator(),
        });

        assert.isTrue(parseResult.result.success);
        expect(parseResult.tree).toMatchSnapshot();
      });
    });
  });

  specify('should not parse in normalized path mode', function () {
    const parseResult = parse('$["\\u0061"]', {
      normalized: true,
      translator: new ASTTranslator(),
    });

    assert.isFalse(parseResult.result.success);
  });
});
