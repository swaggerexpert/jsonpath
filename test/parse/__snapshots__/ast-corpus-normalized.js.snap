// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse ast-corpus-normalized $['\\u000b'] 1`] = `
{
  "type": "JsonPathQuery",
  "segments": [
    {
      "type": "ChildSegment",
      "selector": {
        "type": "NameSelector",
        "value": "",
        "format": "single-quoted",
      },
    },
  ],
}
`;

exports[`parse ast-corpus-normalized $['a'] 1`] = `
{
  "type": "JsonPathQuery",
  "segments": [
    {
      "type": "ChildSegment",
      "selector": {
        "type": "NameSelector",
        "value": "a",
        "format": "single-quoted",
      },
    },
  ],
}
`;

exports[`parse ast-corpus-normalized $['a']['b'][1] 1`] = `
{
  "type": "JsonPathQuery",
  "segments": [
    {
      "type": "ChildSegment",
      "selector": {
        "type": "NameSelector",
        "value": "a",
        "format": "single-quoted",
      },
    },
    {
      "type": "ChildSegment",
      "selector": {
        "type": "NameSelector",
        "value": "b",
        "format": "single-quoted",
      },
    },
    {
      "type": "ChildSegment",
      "selector": {
        "type": "IndexSelector",
        "value": 1,
      },
    },
  ],
}
`;

exports[`parse ast-corpus-normalized $[1] 1`] = `
{
  "type": "JsonPathQuery",
  "segments": [
    {
      "type": "ChildSegment",
      "selector": {
        "type": "IndexSelector",
        "value": 1,
      },
    },
  ],
}
`;

exports[`parse ast-corpus-normalized $[2] 1`] = `
{
  "type": "JsonPathQuery",
  "segments": [
    {
      "type": "ChildSegment",
      "selector": {
        "type": "IndexSelector",
        "value": 2,
      },
    },
  ],
}
`;
